### PR TITLE
Create N-bonacci.cpp

### PR DESCRIPTION
**Pull request by neha-hasija17**
This Pull Request is for HacktoberFest 2020

#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1306"></a>

### CODE REVIEW

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Code is not up to the [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md).
Please read them **carefully** and **follow them**.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Documentation is not up to the [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html).
Please read them **carefully** and **follow them**.

File: math/N-bonacci.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	        /// of the N-Bonacci series
+		std::cout<<"The "<<n<<"-Bonacci series till "<<m<<" terms is"<<std::endl;
+			for(i=0;i<m;i++){
+			std::cout<<a[i]<<" ";
+		}
+		std::cout<<std::endl;
+}
+/// main function takes t as an input for t testcases 
+/// and for each testcase it takes n and m as input and prints N-Bonacci series
+int main() {
```
Comment (Panquesito7) [inline math/N-bonacci.cpp:33]:
```suggestion
/**
 * @brief Main function
 * @returns 0 on exit
 */
int main() {
```

File: math/N-bonacci.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#include <vector>
+/*
+*In general, in N-bonacci sequence, 
+*we generate sum of preceding N numbers from the next term. 
+*For example, a 3-bonacci sequence is the following:
+*0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+*In this code we take N and M as input where M is the number of terms
+*to be printed of the N-bonacci series
+*/
+void N_bonacci(int n,int m){
```
Comment (Panquesito7) [inline math/N-bonacci.cpp:11]: For how to document functions, please see #962 as a reference. 🙂

File: math/N-bonacci.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#include <iostream>
```
Comment (Panquesito7) [inline math/N-bonacci.cpp:1]: For how to document the first comment block, see #962 as a reference.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Documentation is not up to the [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html).
Please read them **carefully** and **follow them**.

**Comment by neha-hasija17** [comment]
@Panquesito7 please check it now, and let me know the error so I can resolve it asap.

File: math/N-bonacci.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (Panquesito7) [inline math/N-bonacci.cpp:1]:
```suggestion
```

File: math/n_bonacci.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
+/**
+ * @file N-bonacci.cpp
+ * @brief Implementation of N-bonacci Series
+ * @author [Neha Hasija](https://github.com/neha-hasija17)
+ */
+#include <iostream>
+#include <vector>
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:7]:
```suggestion
#include <iostream>  /// for std::cout and std::cin
#include <vector>   /// for std::vector
```

File: math/n_bonacci.cpp [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/**
+ * @file N-bonacci.cpp
+ * @brief Implementation of N-bonacci Series
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:3]: Provide a Wikipedia link in Markdown format (if possible).
If there's no Wikipedia link, use another web source for algorithm explanation. 🙂

Comment (neha-hasija17) [reply math/n_bonacci.cpp:3]: http://oeis.org/wiki/N-bonacci_numbers
please refer to above link :)

File: math/n_bonacci.cpp [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ *we generate sum of preceding N numbers from the next term.
+ *For example, a 3-bonacci sequence is the following:
+ *0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ *In this code we take N and M as input where M is the number of terms
+ *to be printed of the N-bonacci series
+ */
+void N_bonacci(int n, int m) {
+    /**
+     * This function finds N-bonacci series for
+     * @param n value and @param m terms and prints the series
```
Comment (Panquesito7) [inline math/n_bonacci.cpp:19]:
```suggestion
     * @param n value
     * @param m terms and prints the series
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/n_bonacci.cpp [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+#include <vector>    /// for std::vector
+/*
+ *In general, in N-bonacci sequence,
+ *we generate sum of preceding N numbers from the next term.
+ *For example, a 3-bonacci sequence is the following:
+ *0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ *In this code we take N and M as input where M is the number of terms
+ *to be printed of the N-bonacci series
+ */
+void N_bonacci(int n, int m) {
```
[the remark this answers is not in the document]
Comment (neha-hasija17) [reply math/n_bonacci.cpp:19]: please help me in resolving this issue

File: math/n_bonacci.cpp [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+ * @returns 0 on exit
+ */
+int main() {
+    int t = 0, n = 0, i = 0, m = 0;
+    std::cout << "Enter number of testcases :";
+    std::cin >> t;
+    while (t--) {
+        std::cout << "Enter the values of N and M : ";
+        std::cin >> n >> m;
+        N_bonacci(n, m);
```
[the remark this answers is not in the document]
Comment (neha-hasija17) [reply math/n_bonacci.cpp:58]: please help me with the same

[the remark this answers is not in the document]
Comment (neha-hasija17) [reply math/n_bonacci.cpp:58]: please help me with the same

[the remark this answers is not in the document]
Comment (Panquesito7) [reply math/n_bonacci.cpp:58]: See the [typical structure](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) of a program as a reference.

File: math/n_bonacci.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (Panquesito7) [reply math/n_bonacci.cpp:19]: See the [typical structure](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) of a program or #962 as a reference.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Comment by Swastyy** [comment]
@neha-hasija17 hey! where did you find the problem statement?
Can you please tell me how can I share my solution?
(P.S. I am new to open source and need some initial help.)

**Pull request closed without merging**